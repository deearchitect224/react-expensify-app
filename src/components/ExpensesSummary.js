import React from 'react';
import numeral from 'numeral';
import { connect } from 'react-redux';
import selectExpenses from '../selectors/expenses';
import selectExpensesTotal from '../selectors/expenses-total';

export const ExpensesSummary = ({expensesCount, expensesTotal}) => {
    const expenseWord = expensesCount > 1 ? 'expenses' : 'expense'
    return (
        <div>
            <h1>You have {expensesCount} {expenseWord} totaling {expensesTotal}</h1>
        </div>

    );
};

const mapStateToProps = (state) => {
    const visibleExpenses = selectExpenses(state.expenses, state.filters);
    return {
        expensesCount: visibleExpenses.length,
        expensesTotal: numeral(selectExpensesTotal(visibleExpenses)).format('$0,0.00')
    };
};

export default connect(mapStateToProps)(ExpensesSummary);
