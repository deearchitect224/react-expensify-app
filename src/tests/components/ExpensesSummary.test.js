import React from 'react';
import { shallow } from 'enzyme';
import { ExpensesSummary } from '../../components/ExpensesSummary';

test('should render Expenses Summary with expenses', () => {
    const wrapper = shallow(<ExpensesSummary expensesCount={5} expensesTotal={10}/>);
    expect(wrapper).toMatchSnapshot();
}); 

test('should render Expenses Summary with empty message', () => {
    const wrapper = shallow(<ExpensesSummary expensesCount={0} expensesTotal={0}/>);
    expect(wrapper).toMatchSnapshot();
}); 

test('should render Expenses Summary with one expense', () => {
    const wrapper = shallow(<ExpensesSummary expensesCount={1} expensesTotal={50}/>);
    expect(wrapper).toMatchSnapshot();
}); 