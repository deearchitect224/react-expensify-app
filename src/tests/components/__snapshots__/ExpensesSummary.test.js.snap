// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`should render Expenses Summary with empty message 1`] = `
<div>
  <h1>
    You have 
    0
     
    expense
     totaling 
    0
  </h1>
</div>
`;

exports[`should render Expenses Summary with expenses 1`] = `
<div>
  <h1>
    You have 
    5
     
    expenses
     totaling 
    10
  </h1>
</div>
`;

exports[`should render Expenses Summary with one expense 1`] = `
<div>
  <h1>
    You have 
    1
     
    expense
     totaling 
    50
  </h1>
</div>
`;
